### DISCUSSION TITLE
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ""

### DISCUSSION THREAD

**Discussion by plus-** [TypeStrong/ts-node#1911] [Q&A]
Hello,

We're migrating a project in Node 18, this project uses ts-node to execute sequelize script in the following manner:

`yarn ts-node-esm --require dotenv-flow/config $(yarn bin)/sequelize`

Since migrating to Node 18 we see the following error:

> /application/node_modules/ts-node/dist-raw/node-internal-modules-esm-get_format.js:93
>         throw new ERR_UNKNOWN_FILE_EXTENSION(ext, fileURLToPath(url));
>               ^
> CustomError: ERR_UNKNOWN_FILE_EXTENSION  /application/node_modules/sequelize-cli/lib/sequelize
>     at defaultGetFormat (/application/node_modules/ts-node/dist-raw/node-internal-modules-esm-get_format.js:93:15)

We've tried fixing this error with the `experimental-specifier-resolution` option without success:

`yarn ts-node-esm --experimental-specifier-resolution=node --require dotenv-flow/config $(yarn bin)/sequelize`

>  ExperimentalWarning: The Node.js specifier resolution in ESM is experimental.
> (Use `node --trace-warnings ...` to show where the warning was created)
> TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension "" for /application/node_modules/sequelize-cli/lib/sequelize
>     at new NodeError (node:internal/errors:393:5)
>     at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:79:11)

We've seen this [update](https://github.com/TypeStrong/ts-node/pull/1838/files) in the last version of ts-node and read the [issue](https://github.com/nodejs/node/pull/43763) on the node repo without being able to find a solution using ts-node.

Any idea how we can fix this or if this is going to be fixed in the future in node 18?

**Comment by bonham** [comment]
This tsconfig.json works for me with node 18.16.0

tsconfig.json
```
{
  "compilerOptions": {
    "target": "ES6",
    "module": "Node16",
    "outDir": "dist",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  },
  "ts-node": {
    "esm": true
  }
}
```

package.json
```
{
  "name": "tsnodepoc",
  "version": "1.0.0",
  "main": "index.ts",
  "type": "module",
  "scripts": {
    "build": "tsc",
    "start": "ts-node index.ts"
  },
  "devDependencies": {
    "ts-node": "^10.9.1",
    "typescript": "^5.0.4"
  }
}
```
